### PR TITLE
feat: v-show

### PR DESCRIPTION
**Pull request by zhangmo8**
Supports `v-show` in vapor

### CODE REVIEW

File: packages/compiler-vapor/src/transform.ts [hunk 1]
```diff
@@ -444,6 +444,20 @@ function transformProp(
       })
       break
     }
+    case 'show': {
+      if (expr === null) {
+        // TODO: support v-show
+        return
+      }
+
+      ctx.registerEffect(expr, {
```
Comment (baiwusanyu-c) [inline packages/compiler-vapor/src/transform.ts:453]: Does this implementation require further design and discussion @sxzz? Currently, vapor lacks some runtime design. If it needs to be functionally unified with vue/core, then it needs a life cycle design, such as the life cycle of instructions and the life cycle of components. In vue/core v-model and v-show rely on the life cycle of instructions for implementation. I am not sure whether this is needed in vapor.

Comment (zhangmo8) [reply packages/compiler-vapor/src/transform.ts:453]: Emm I will make some changes according to the design after @sxzz to reply.

Comment (sxzz) [reply packages/compiler-vapor/src/transform.ts:453]: Sure, we need to implement basic runtime directives first
See https://vuejs.org/guide/reusability/custom-directives.html#directive-hooks

**Comment by zhangmo8** [comment]
Waiting for Component `Lifecycle` & `Transition` component

**Comment by sxzz** [comment]
Thanks for your contribution, but we should use the custom directive for `v-show`. Closing this because the way of runtime implementation is completely different.

Welcome to contribute again once the `updated` hook works.

**Pull request closed without merging**